src/bin/rustfmt.rs

```diff
@@ -119,67 +115,33 @@ fn match_cli_path_or_file(config_path: Option<PathBuf>,
     resolve_config(input_file)
 }
 
-fn update_config(config: &mut Config, matches: &Matches) -> Result<(), String> {
+fn update_config(config: &mut Config, matches: &Matches) -> Result<()> {
     config.verbose = matches.opt_present("verbose");
     config.skip_children = matches.opt_present("skip-children");
 
-    let write_mode = matches.opt_str("write-mode");
-    match matches.opt_str("write-mode").map(|wm| WriteMode::from_str(&wm)) {
```

Comment:
This refactoring doesn't seem to be an improvement to me. Admittedly, the original match was a bit complex, but now we have two nested `if let`s, an early return, and a separate return value, which seems worse.